packages/wrangler/src/dev/start-server.ts

```diff
@@ -72,6 +74,34 @@ export async function startDevServer(
 		}
 	}
 
+	const devEnv = new DevEnv();
```

Comment:
This outer function (startDevServer) is called internally by unstable_dev. Here we create a DevEnv and fake events to the ProxyController by calling it's event handler methods (`devEnv.proxy.on*`)